test/test_gpu_view.cpp

```diff
@@ -2449,4 +2449,84 @@ TEST_F(NVFuserTest, ReshapeOfReshape_CUDA) {
   TORCH_CHECK(ref.equal(cg_outputs.at(0)));
 }
 
+TEST_F(NVFuserTest, ReshapePermuteTransposeScheduler_CUDA) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  std::vector<int64_t> shape({256, 128, 64});
+
+  auto tv0 = makeSymbolicTensor(3);
+  fusion->addInput(tv0);
+
+  auto tv1 = reshape(tv0, {256, 128, 64}, {256, 128, 8, 8});
+  auto tv3 = transpose(tv1, 1, 3);
+  fusion->addOutput(tv3);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+
+  auto t0 = at::randn(shape, options);
+  std::vector<c10::IValue> aten_inputs({t0});
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+  auto cg_outputs = executor_cache.runFusionWithInputs(aten_inputs);
+
+  auto runtime = executor_cache.getMostRecentKernelRuntime();
+  TORCH_CHECK(!runtime->isSegmented(), "Segmentation not expected");
+
+  auto heuristic =
+      runtime->schedulerHeuristics()->heuristicsList().at(0).get()->heuristic();
+  TORCH_CHECK(
+      heuristic == ScheduleHeuristic::Transpose,
```

Comment:
:vomiting_face: 

Turns out this example that we do want to support actually does involve a small transpose dimension. I guess this isn't surprising at all.
So there's no easy way out from just disabling view + small transpose dimensions. I need to actually look at the relationship between the reshape ops and the split/merge necessary from transpose scheduler?!

Reply:
ok this is a false alarm.

I mean, we still do want to enhance our transpose scheduler, but the test case in nanogpt is actually in a different shape and doesn't trigger small transpose dimension.